src/DocumentEditor.ts

```diff
@@ -4,154 +4,79 @@
+        const docLink = node.getSelfLink();
+        output.appendLine(`${timestamp}: Updated entity "${docLink}"`);
         output.show();
-        await this.updateEditor(updatedDoc);
+        await this.updateEditor(updatedDoc, vscode.window.activeTextEditor);
     }
 
-    private async updateEditor(data: {}): Promise<void> {
-        const textEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor;
-        if (textEditor && this.isLocalDocPath(textEditor.document)) {
-            await util.writeToEditor(textEditor, JSON.stringify(data, null, 2));
-
-            this.ignoreSave = true;
-            try {
-                await textEditor.document.save();
-            } finally {
-                this.ignoreSave = false;
-            }
+    private async updateEditor(data: {}, textEditor: vscode.TextEditor): Promise<void> {
+        await util.writeToEditor(textEditor, JSON.stringify(data, null, 2));
+        this.ignoreSave = true;
+        try {
+            await textEditor.document.save();
+        } finally {
+            this.ignoreSave = false;
         }
     }
 
     public async onDidSaveTextDocument(globalState: vscode.Memento, doc: vscode.TextDocument): Promise<void> {
-        if (!this.ignoreSave && this.isLocalDocPath(doc) && this.lastOpenedDocNode) {
-            // soft-copy the node to avoid race conditions
-            const node: IDocumentNode = this.lastOpenedDocNode;
-
-            // If the user saved the file as a part of closing VS Code, we might not be able to 'Update to Azure' in time
-            // However, we can copy the file to this path and prompt user about it on next activation
-            this.recoveredFileName = `${this.lastOpenedDocNode.label}${Date.now().toString()}.json`;
-            try {
-                const dontShow: boolean | undefined = globalState.get(this.dontShowKey);
-                if (dontShow !== true) {
-                    const message: string = `Saving "cosmos-document.json" will update the CosmosDB document "${node.label}" in Azure.`;
-                    const result: string | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.OK, DialogBoxResponses.DontShowAgain);
-
-                    if (!result) {
-                        throw new UserCancelledError();
-                    } else if (result === DialogBoxResponses.DontShowAgain) {
-                        await globalState.update(this.dontShowKey, true);
-                    }
+        const filePath = Object.keys(this.fileMap).find((filePath) => path.relative(doc.uri.fsPath, filePath) === '');
+        if (!this.ignoreSave && filePath) {
+            const node: IEditableNode = this.fileMap[filePath][1];
+            const dontShow: boolean | undefined = globalState.get(this.dontShowKey);
+            if (dontShow !== true) {
+                const message: string = `Saving this file will update the entity "${node.label}" in Azure.`;
```

Comment:
'cosmos-editor.json' instead of 'this file'
'to the Cloud' instead of 'in Azure'